### DISCUSSION TITLE
next/image Styling

### DISCUSSION THREAD

**Discussion by jillmolloy** [vercel/next.js#18312] [Help] [answered]
I'm working on replacing images in an existing web app with the new Image component, but I've immediately hit a bit of an issue as far as styling is concerned.

For example, I have an image in my app that's being centered with `margin: 0 auto;` being applied to the image. When I swap out the default `<img>` element for `<Image>` it loses that centering. From inspecting I can see the image is now being wrapped in a number of `<div>` elements. So, I then attempted to apply a `className` to the Image, hoping it would get passed to the parent `<div>` so that I could center that, but it seems that this is not being passed to the element.

My current workaround is to now wrap the `<Image>` element in another `<div>` and apply the styles to that, but I feel like this increases the level of effort for upgrading quite a bit across the entire site, between adding new wrappers and new styles.

I was wondering if there was a more straightforward way to achieve this, perhaps it's as simple as allowing the `className` prop for the Image component? Or perhaps there's an even better solution already in place that I've missed entirely – if so, apologies for this pre-emptive issue.

Many thanks for all the work you guys do – big fan!

**Comment by petermanser** [comment]
`className` is allowed for the `Image` component, but that doesn't solve the problem.

**Reply by jillmolloy** [reply]
Sorry @petermanser, you're correct, and I was not clear that it does pass `className` down to the underlying image, rather than the parent div of the Image component and that does not solve the problem. 

I think styling the parent and accessing the underlying image via Sass / CSS would be more helpful here.

**Reply by reckter** [reply]
You would need a `wrapperClassName` so you can apply stylings to the `div`s sourounding the `img`.
I would even argue, that the default of `className` should be, to be aplied to the `div`, and an additional `imgClassName` that actually applies the class to the `img` component

**Reply by jillmolloy** [reply]
I agree @reckter, that's what I was suggesting as you can still target the child image of the wrapper via Sass/CSS.

**Reply by uinstinct** [reply]
Neither `className` nor does `style` works

**Reply by sethwebster** [reply]
For apps where you'd like to offer some dynamic image styling (for example, a hero image that needs a specific `object-position`), it would be very useful to be able to pass dynamic styles.

**Reply by uinstinct** [reply]
@sethwebster Can you please show me a demo of image styling of _next/image_? I will find it very useful

**Reply by guitheengineer** [reply]
There's one more problem, even if you added classNames for the wrapper, it still has inline styles which would have priority over your CSS

**Reply by guitheengineer** [reply]
So for example if you add a wrapperClassName and set it's position to absolute, won't work because it already has an inline style with position: relative that take the precedence. The only way I know to take classnames priority over inline styles is by using !Important, but as we know isn't a recommended way of doing it.

**Reply by guitheengineer** [reply]
Maybe the cleanest way of solving It would be adding an optional styleWrapper prop for the wrapper and a style for the img, like this
```jsx

export default function Image({...otherNextjsProps, styleWrapper, style)} {
  const wrapperStyle = {/* nextjs styles */}
  return (
    <div style={{...wrapperStyle, ...styleWrapper}}>
      <img style={{...nextjsImgProps, ...style}} />
    </div>
  )
}
```

Is possible to make it work with classnames, but all nextjs styles won't be applied
```jsx
export default function Image({...otherNextjsProps, wrapperClassName, className)} {
  const wrapperStyle = {/* nextjs styles */}
  return (
    <div style={{!wrapperClassName ? wrapperStyle : undefined}}>
      <img style={{!className ? nextjsImgProps : undefined}} />
    </div>
  )
}
```

**Comment by saintjcob** [comment]
className works, unless the classes are added via CSS Modules. Then it adds "undefined", couldn't find the workaround.

**Reply by uinstinct** [reply]
what about `style jsx`?

**Reply by saintjcob** [reply]
Haven't tried that yet. I'd like to keep one way of styling as much as possible. For now I'm using a div.imageWrapper as a workaround, and it works well

**Reply by uinstinct** [reply]
Can you show me your way of using an `div.imageWrapper` to style the _next/image_?

Also, is there a way to use _next/images_ during `next export`? (because I see blank images when doing next export)

**Reply by saintjcob** [reply]
Sure, i did it like that:

`<div className={styles.imageWrapper}>
                <Image
                  src={path}
                  alt='Picture of the "Shape 1" painting'
                  width={593}
                  height={892}
                />
              </div>`

Then in my component's module.css file I styled it as usually:

`.imageWrapper {
      display: flex;
      flex-direction: column;
      margin-bottom: 16px;

      &:last-of-type {
        margin-bottom: 24px;
      }
    }`

Can't tell you anything about next export, as I haven't used it yet.

**Reply by MauricioGedrat** [reply]
adicionem o atributo: layout="responsive". A imagem preenchera o container

**Comment by jawadakram20** [comment]
Class names not working for me either. Tried styles, classNames but nothing works. I wanted to add a border-radius to the image.

**Reply by RamziBach** [reply]
Exact same thing, I am trying to add a border-radius but I can't seem to figure out a way to do it.. Which is CRAZY to think about.

EDIT: I found a temporary solution until this is resolved, I had to make a global.css and import it to _app and was able to select the className from my next/image. Essentially, next/image has issues with CSS in JS and CSS Modules. Other then that it works as intended.

**Reply by hanzlaharoon** [reply]
With `bootstrap` simply add `rounded circle` class.

```html
<Image className="rounded-circle" src="./whatever.jpg" width={30} height={30} />
```
hope this helps!

**Comment by stephan281094** [comment]
Stumbled upon this as well when using `styled-jsx`. It works when making it global:
```jsx
<>
  <Image className="avatar" src="./whatever.jpg" width={30} height={30} />
  <style jsx global>{`
    .avatar {
      border-radius: 50%;
    }
  `}</style>
</>
```

**Reply by maythanhsuaung0-0** [reply]
that works!Thank you man

**Reply by ezraamos** [reply]
life saver haha

**Reply by ezraamos** [reply]
doesnt work if you add border e.g solid white 5 px

**Reply by stephan281094** [reply]
> doesnt work if you add border e.g solid white 5 px

Adding `!important` works, but might not be the ideal solution.

**Reply by Lmraza98** [reply]
Any solution for adding a border to Next/Image? I've been struggling to get this to work, even with '!important' included. Including the className and styled block worked for border-radius though.

**Reply by guitheengineer** [reply]
> 
> 
> Any solution for adding a border to Next/Image? I've been struggling to get this to work, even with '!important' included. Including the className and styled block worked for border-radius though.

Add a div with the border above the Image component.

```
<div style={{ border: '2px solid red' }}>
    <Image />
</div>
```

**Reply by Lmraza98** [reply]
@guilhermefront So I did try this, unfortunately the border is off by a couple of pixels making it look pretty weird. I'm thinking I may just need to use a generic img component. I'm trying to turn my image into a circle with a border.

**Reply by Lmraza98** [reply]
<img width="412" alt="Screen Shot 2021-07-03 at 12 09 56 PM" src="https://user-images.githubusercontent.com/38518956/124360415-a7969200-dbf7-11eb-816f-d630a3d988ff.png">

**Reply by zontyp** [reply]
hi

did you change  <style jsx> to  <style jsx global>
?

**Reply by zontyp** [reply]
it worked, but not sure what global means, will study it later or if someone can answer.

**Reply by stephan281094** [reply]
> it worked, but not sure what global means, will study it later or if someone can answer.

By making something global you disable scoping, see https://github.com/vercel/styled-jsx#global-styles.

**Reply by spandita46** [reply]
> 
> 
> <img alt="Screen Shot 2021-07-03 at 12 09 56 PM" width="412" src="https://user-images.githubusercontent.com/38518956/124360415-a7969200-dbf7-11eb-816f-d630a3d988ff.png">

I know it's late, but might save somebody some time. 
To fix the **"border being off by few pixels"**, try making the parent wrapper of image on which border colour is applied to be of display:inline-flex.  Codesandbox - https://codesandbox.io/s/nextjs-avatar-component-q2rcr?file=/components/avatar.module.css

**Reply by hanzlaharoon** [reply]
With `bootstrap` simply add `rounded circle` class.

```html
<Image className="rounded-circle" src="./whatever.jpg" width={30} height={30} />
```
Read more at docs [Bootstrap - Border Radius](https://getbootstrap.com/docs/5.0/utilities/borders/#border-radius)

**Reply by fsgreco** [reply]
thank you @spandita46 , you indeed save me time!

**Reply by edgarfb** [reply]
Thanks bro. Also works if you create a class like  .rounded-circle on the globals.css and then uses it on <Image className="rouded.circle"/>

**Comment by jeremyzilar** [comment]
So why does Next automatically add those `<div>`s around the `<img` with the additional css?

**Reply by T04435** [reply]
I think, it is to prevent layout shifting.
The `divs` will take the space the image would after it loads.

**Reply by muffeeee** [reply]
To make using it a complete nightmare. I don't get what the point is to improve all these performance metrics when it is literally impossible to use in any project in a sensible way. The fact that _everyone_ has to develop their own "work-around" to use this component is just mind boggling.. A prime example of over-engineering indeed

**Reply by Amar-Gill** [reply]
I'm trying to add a border to my image. I can't simply do `className='border-solid border-slate-400 border-2'` (tailwindcss). I have to use the important override `className='!border-solid !border-slate-400 !border-2'`.

**Comment by juniorbatistadev** [comment]
having the same issue trying to style an image using css modules

**Reply by zontyp** [reply]
did you use  <style jsx global>?

**Comment by anshuopinion** [comment]
```
import styled from "@emotion/styled";
import React from "react";

const App= () => {
  return (
   <Box >
          <StyledImage
            src="image/url"
            height="160"
            width="180"
            objectFit="cover"
          />
        </Box>
  );
};

export default App;

const Box = styled.div``;

const StyledImage = styled(Image)`
  border-radius: 10px;
`;

```

Try this code...

Use styled-component or o@emotion to modify image component of next js

**Reply by ganeshkarthik-n** [reply]
this approach didn't work for me and I had to put the image inside a wrapper div.

**Reply by JesusHV** [reply]
It works for me with next `10.2.2`

**Reply by rlenoir-codepoint** [reply]
Thank you @anshuopinion , an example using styled-components & Typescript

```typescript
const StyledImage = styled(Image)<{ $borderRadius: string }>`
  border-radius: ${(p) => p.$borderRadius};
`;

<StyledImage src="/foo/bar.jpg" layout="fill" objectFit="cover" alt="FooBar" $borderRadius="6px 6px 0 0" />
```

**Comment by guitheengineer** [comment]
Easiest solution (as you said) is by simply wrapping the `<Image />` component and styling it. The problem is simply not having a prop to style the div container that `<Image />` has.
```jsx
<div className={styles.image}>
  <Image width={500} height={500} src="assets/image.png" />
</div>
```
If you want to style the img element itself, classes are working fine, the problem is that img has internal inline styles that will have a higher CSS priority than yours. This is the CSS that overwrites:
```css
visibility: inherit;
position: absolute;
top: 0px;
left: 0px;
bottom: 0px;
right: 0px;
box-sizing: border-box;
padding: 0px;
border: none;
margin: auto;
display: block;
width: 0px;
height: 0px;
min-width: 100%;
max-width: 100%;
min-height: 100%;
max-height: 100%;
```
You can use !important (`property: value !important;`) in your css and it will work (i'm not sure of the bad consequences of this)

**Reply by osdiab** [reply]
i agree - i think it's an antipattern that the `className` actually styles the child image, not the parent div.

Here's an example of unexpected behavior: if you have a flex parent like so (i'm using emotion in this example, `css` prop just sets `className` with those styles):

```
<div css={{ display: "flex" }}>
  <Image css={{ flexShrink: 0 }} width={400} height={400} />
  <p>Lorem ipsum dolor sit amet...</p>
</div>
```

then trick's on you, flex shrink is applied to the inner child and then has no effect since the div wrapper isn't a flexbox element. The fact this doesn't work is directly a result of `className` targeting the wrong element.

In my opinion what would make sense is to expose both `className` and `imgClassName`, where the `className` param as expected targets the wrapper `div`. Then we're peachy. But that's a breaking change, so in the meantime I'll just export my own wrapper in my project that wraps the wrapper *facepalm*

**Comment by zraineri** [comment]
Was enjoying this until I tried styling the images, seems like a lot of extra effort to convert any website over

**Reply by shivu-dg-nuzpapr** [reply]
So true :'( Why do I have to go through this!!
I'm migrating a react app to nextjs which used scss for styling, now all the scss styling is being overridded by this stuping nextjs Image tag 😥!!.  sorry for the ranting

**Reply by pjimola** [reply]
I had literally the same thought. And now it's mid-2022, are we for real still dealing with that?! smh

**Comment by AmmarHalees** [comment]
Any updates guys? I'm using css modules and trying to keep component composition.

**Reply by landed1** [reply]
I'm getting so many issues styling images I am considering dropping the component!

**Reply by AmmarHalees** [reply]
Same here. next/image isn't doing anything magical. Most of the functionality it provides can be done natively.

**Reply by lveillard** [reply]
did you guys find a good alternative to optimize images?

**Comment by diogosilva95** [comment]
Same here, next/image isn't any good when it comes to use them with custom positioning and dimensions. Maybe Next team can offer an option to use the compression to webp only.

**Comment by jonrrivera** [comment]
lol why is it so hard to style this?

**Reply by coreymunn3** [reply]
There's so much else going on that they're trying to account for with the wrapper divs. It really is annoying though.

**Comment by ghost** [comment]
Same problem, I just need to add border and border-radius for the image without wrappers or !important

**Reply by garfiaslopez** [reply]
hahaha sameee.

**Comment by dwightee** [comment]
This is stupid, the default behavior is just impossible to work with. I have just spend couple hours to make it work but I am dropping it. My main issue was to make the images naturally spread in the remaining space. Simple change to `<img />` and everything works. 
There is a div with styling that just completely crashes it (mentioned above).

**Reply by ahmedsedeeek3** [reply]
but we  still need the performance

**Reply by yanickrochon** [reply]
The `Image` component is wrapped by an absolute positioned `div`... why?? This breaks the flex layout and I have to fiddle with styles in order to undo the crap that's added with the component. Helper components should let users manage their own layouts.

**Reply by pjimola** [reply]
I've tried already to change from Image to img, but still my class isn't in the DOM

**Comment by estebanmunchjones2019** [comment]
Here is a solution that works for accessing the first internal wrapping div of the `<Image>` element
```
// JSS styling here //
imageWrapper: {
    "& div:first-child": {
      position: "static !important"
    }
  }
```

```
//JSX code here//   
<div className={classes.imageWrapper}>
   <Image src={image1} layout="responsive" width={1450} height={750} alt="First slide" />
</div>
```

**Reply by wfedev** [reply]
How can I apply the default style without using !important?

**Reply by ahmedsedeeek3** [reply]
.img > :nth-child(1) {
	border-radius: 50% !important;
}
 this how i solved my problem ...i need to use it as avatar

**Reply by destronogen** [reply]
This worked for me. Thank you @estebanmunchjones2019

**Comment by nkrowicki** [comment]
### Here my temporal solution to add rounded border with padding

⬇

https://github.com/vercel/next.js/discussions/22861#discussioncomment-735713

**Comment by pedrodimarco** [comment]
I didn't have any problem with border-radius propertie, but a I still can't set background-size and other background properties on the image component

**Comment by wfedev** [comment]
I checked this on Discord & Nextjs Issues

1. There is also a way to apply it using imputation.
[layout fill](https://github.com/vercel/next.js/blob/canary/examples/image-component/pages/layout-fill.js)

2. Modify image component examples app for static image
https://github.com/vercel/next.js/pull/25956

in another way

https://nextjs.org/docs/basic-features/image-optimization#image-sizes

I think !important has its limits. @estebanmunchjones2019

**Comment by alexmk92** [comment]
It's a fantastic bit of kit but I would love for it to be headless...the default styles are just a nightmare to work with due to there being an inline style tag embedded on the `<img>` 

Using `ImageKit` and my own Image component for the time being, I'll be coming back to this at some point I'm sure.

**Comment by ahmedsedeeek3** [comment]
.img > :nth-child(1) {
	border-radius: 50% !important;
}
 you may use the targeting  method to over write  the default CSS of image

**Comment by robFraser1111** [comment]
I fixed this by adding the !important flag to my css properties.

**Reply by juniorbatistadev** [reply]
Not ideal but I guess it do the job

**Comment by ghost** [comment]
This component is truly something, I'd much prefer Next not trying to force some extremely specific layout that is filthy and problematic to override and just render an img with a srcset. Maybe export another slimmer component so there's a possibility to avoid this one. If at least the containers were styled well, right now it doesn't even respect parent height.

**Reply by galago** [reply]
Same issue here, a trivial task like fit an image to its parent height seems impossible

**Reply by alexecus** [reply]
I totally agree. Is it that hard to put a flag on Next/Images to stop putting inline styles at user's will ? I also don't get the point of providing an image wrapper then enforcing their sets of style

**Reply by juniorbatistadev** [reply]
Completely agree!

**Reply by lveillard** [reply]
+10000

**Comment by juniorbatistadev** [comment]
I'm surprised they haven't solved this.

**Reply by miyamoto-san** [reply]
You're probably more than welcome to submit a PR you know :)

**Reply by yanickrochon** [reply]
There are a few who already have, but they seem to be ignored.

**Comment by lveillard** [comment]
I have the same issue with tailwind. Next/image just ignores everything i add to its className.
Also it forces me to give width and height. Why adding extra requirements? native img does not ask for that

I'm trying to add these clases and non of them work:
                className="order-first lg:order-last mb-8 lg:mb-0 max-h-96 max-w-3/4 mx-auto"

**Reply by ghost** [reply]
That's because the classes are sent to the img element inside the two div wrappers that the Image component renders with. You can work around it by wrapping the Image component with another element and write selectors that overpower the div styles (with `!important` as those have inline styles). Absolute filth obviously, but that's on the authors of this component.

**Reply by lveillard** [reply]
Thanks for the tip!
I think I will have the images unoptimized for now and wait for a fix as I don't want to add css classes and ugly !important, 
I'm trying to have everything managed by tailwind exclusively

I wish i was good enougt to propose a good PR to fix it myself, it seems to be impacting a lot of people!

**Reply by ghost** [reply]
I think you can also hack around it a little bit more "cleanly" by invalidating those wrappers with `display: contents`, so their styles will not influence the img tag inside, which is what I decided to do. That will obviously kill some of the features the component creates, but in most cases you probably don't even want them since they mostly translate into styles you can just do yourself.

**Reply by lveillard** [reply]
I've added the contents tag and neither adding width+height or layout="fill" + that display: contents fixes it. Tailwind classes are keep ignored :(

I think they have a div inside a div and then the img.

**Reply by lveillard** [reply]
confirmed it's a div inside a div and then the img
![image](https://user-images.githubusercontent.com/29461076/134550408-bea807e7-8677-4e9e-93c7-9408330374b7.png)

**Reply by lveillard** [reply]
If only there was a way to "unwrap" a component.

```
import Image as Wrapper from 'next/image'
const RealImage = Peel( Wrapper, 2) 
```

=>And voilá we could style that img 😂

**Comment by lveillard** [comment]
I've just created a feature request, feel free to upvote it :)
https://github.com/vercel/next.js/issues/29339

**Comment by Ayushthapa123** [comment]
In my case placing the position of the parent element to relative solve the problem. 
It is in sass format.

    .cart {
       width:25%;
       height:300px;
       background-color: whitesmoke;
       margin:10px;
       position: relative;
       .image {
           width:100%;
           height:200px;
           object-fit: cover;
       }
 
    }

**Comment by pawelpikus** [comment]
As Peter Lunch nicely explains, the Nextjs Image component nests img into couple of other containers. In his blog post at https://www.peterlunch.com/snippets/next-image-styling, he is talking about div, but I've got my img wrapped into spans. So, styling was possible using:

.img_wrapper > span {
  box-shadow: $base-box-shadow;
}

Hope it helps. Cheers.

**Comment by estebanmunchjones2019** [comment]
hi everyone!
This article explains the process of creating that Next.js <Image /> component, by the team who created it.
https://web.dev/image-component/
It helped me to understand the idea behind that component

**Comment by lveillard** [comment]
Thanks for the article @estebanmunchjones2019 
Tailwind 3.0 (currently in alpha) will have "aspect-ratio" so we should be able to use it.

I hope we can follow the ideas on that document and stop forcing height and width so we can also use aspect-ratio 

Also last time I tried to style the component Image with tailwind, it was still not possible (nextjs 12)

**Comment by angusryer** [comment]
@estebanmunchjones2019 --> This a great explanation of performance-related considerations, thank you! I agree with all of them. However, when the development experience using this component is trying enough that many throw the component away in place of something easier to use--then the performance points are lost.

This line in the article:

> Since developers cannot use the Image component unsized, the design **ensures that they will take the time to consider image sizing** and prevent layout shifts.

In other words, the solution they devised to make sure that layout remains performant was to force developers into accommodating the component designer's opinions about how to solve the "unknown image size" problem. I mean no malice there at all--the component is still doing a ton of optimization work that would take me months to figure out and implement. But admittedly I don't  "take the time to consider image sizing" enough to provide hard-coded sizes, and having to write abstruse and !important CSS just to do the most common use-case styling complicates my code and feels like an anti-pattern.

Image sizes are often unknown to developers--so at least the `layout` prop should behave predictably, which it does not... yet.

Not sure I'm confident enough to jump in with a PR, but I'll have a look. 🙃

**Reply by osdiab** [reply]
I get why they made it like they did but there's just bad design decisions in how the component works, namely that className doesn't set the className of the outermost element, but rather an inner one, making it basically impossible to style it in a standard way in the context of a layout without using CSS selectors that exploit knowledge of the internals of how next/image works. Honestly just that one change would probably make it a lot more usable for me.

**Reply by angusryer** [reply]
You may be right. I wonder what implications that would have on the existing styling decisions. The component styles are in-line -- did they do that for specificity to ensure that the sizes or layout provided as props absolutely stay that way? If so, then maybe lowering that specificity will break design performance characteristics.

Rhetorical questions! I haven't looked at the code base yet.

**Comment by johnzenith** [comment]
Styling images on Next.js is one thrilling outcome. After doing lots of internal module CSS and css-in-js, I decided to move to import styles from components to global CSS, since am using Tailwindcss.

// global.css
// tailwindcss rules

// components styles
// @import '/components/Image/style.css'

**Comment by theHammi01** [comment]
I've spent 3 hours debugging this, absolutely unbelievable

**Reply by johnzenith** [reply]
just target the image using component module CSS (give the parent element you wrapped the image in a class attribute):

Note: Images are styled in next.js using nested span elements.

.imageWrapper > span {
  width: 100px;
  height: 100px;
  ...
}

**Comment by lveillard** [comment]
Also it makes no sense to force devs to use one styling methodology in particular. The fact that this does not work with Tailwind is so frustrating

**Reply by corydhmiller** [reply]
Chiming in here that the way <Image /> handles styling and does not allow for utility classes to override the inline styles that the image generates is frustrating almost to the point of wanting to not use <Image /> at all. For instance, I would like to round the image and add a simple light green border.

`<Image src={...} className="rounded-full border border-solid border-green-200" width={300} height={300} />`

Of course, this doesn't work, because the classes are dropped onto the `<img>` element and thus are overridden by the myriad of inline styles that are added by the Image component.

**Comment by lveillard** [comment]
Hello @ijjk 👋
This is one of the top discussions in next.js, but it does not look like a discussion, everyone seems to be aligned. Do you think we could re-open this issue: https://github.com/vercel/next.js/issues/29339 that you have closed or another one so at some point this gets into the backlog?

**Reply by jellohouse** [reply]
Please let's get this fixed! 
I'm trying to dynamically add a random background color to my image with inline react styles, and it's super complicated. Still trying to figure it out... it's ridiculous.

**Comment by galago** [comment]
Since v12.1.1 there's a workaround with` layout = "raw"`, which doesn't add any wrap or styling but still lets you use the rest of the features.

Please check the reference:
https://nextjs.org/docs/api-reference/next/image#experimental-raw-layout-mode

**Reply by lveillard** [reply]
Apparently, is still useless with tailwind. If you want to do something like this:
```
className="w-44 md:w-56 lg:w-64"
```
you will need to add those in JS to the height and width properties of image

**Reply by dstroot** [reply]
Why not just create a div with the classes above and having the image fill it?

**Reply by jaxramus** [reply]
@dstroot  because its a terrible DX

**Reply by lack-of-gravitas** [reply]
and @galago 's suggestion isnt? when tailwind use is widespread

**Comment by yanickrochon** [comment]
This component should be refactored and Next should provide a better implementation with a transition roadmap. The current implementation should become deprecated as it is not only frustrating, it is a bad design.

**Comment by CodeAnnihilator** [comment]


**Reply by lack-of-gravitas** [reply]
useless docs for a new adopter

**Comment by DenisBun** [comment]
It's been 2 years since that discussion was opened. Just wondering how you guys handle it. I've been trying both default and future <Image /> components and the layout is just ripped off. Don't really want to spend month fixing each image case :(

**Reply by lack-of-gravitas** [reply]
we're managing it like code cannibalized from elsewhere -- i.e. if it works, isolate it and don't touch it and pray for the best... vercel are a good crew but not providing good doco for this AND then crippling deployments because my code has legacy img tags is just not cool...

**Comment by dchien234** [comment]
It's really sad that this topic doesn't receive enough attention from project's maintainers. If I use `<Image>` component with TailwindCSS or any CSS styling, NextJS will override the `className` value with a bunch of opinionated classes, and all my image style will break.

Please fix this if you want people to use/utilize the advantages that `<Image>` component has to offer.

**Reply by osdiab** [reply]
They made a new image component with next 13, though I haven’t used it before so idk if it fixes these problems.

**Comment by leerob** [comment, marked as answer]
Hi folks, this discussion was originally opened over 2 years ago.

Since then, we've made _a lot_ of improvements to `next/image` directly from your feedback, including removing the wrapping elements (making it easier to style) as mentioned above.

Please try out the new Image component in v13: https://nextjs.org/blog/next-13#nextimage

I'm going to close out this discussion, but feel free to open a new discussion if you're having trouble with the new image component. Thank you!
